tests-utils/src/main/scala/tech/beshu/ror/utils/containers/windows/WindowsPseudoContainer.scala

```diff
@@ -0,0 +1,66 @@
+/*
+ *    This file is part of ReadonlyREST.
+ *
+ *    ReadonlyREST is free software: you can redistribute it and/or modify
+ *    it under the terms of the GNU General Public License as published by
+ *    the Free Software Foundation, either version 3 of the License, or
+ *    (at your option) any later version.
+ *
+ *    ReadonlyREST is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU General Public License for more details.
+ *
+ *    You should have received a copy of the GNU General Public License
+ *    along with ReadonlyREST.  If not, see http://www.gnu.org/licenses/
+ */
+package tech.beshu.ror.utils.containers.windows
+
+import org.testcontainers.containers.GenericContainer
+
+import java.util.concurrent.atomic.AtomicReference
+
+abstract class WindowsPseudoContainer[T <: GenericContainer[T], SERVICE] extends GenericContainer[T]("noop:latest") {
+
+  protected def name: String
+
+  protected def prepare(): SERVICE
+
+  protected def awaitReady(): Unit
+
+  protected def destroy(service: SERVICE): Unit
+
+  protected def getPort(service: SERVICE): Int
+
+  private val service: AtomicReference[Option[SERVICE]] = new AtomicReference(Option.empty[SERVICE])
+
+  override final def start(): Unit = doStart()
+
+  override final def doStart(): Unit = {
+    val success = service.compareAndSet(None, Some(prepare()))
```

Comment:
It's dangerous to do it like that. The `compareAndSet` may be called many times, and we have potential side effect in the `prepare` method.